x-pack/test/fleet_api_integration/apis/epm/get.ts

```diff
@@ -182,6 +182,9 @@ export default function (providerContext: FtrProviderContext) {
       const dataStream = packageInfo?.data_streams?.find(
         ({ dataset }) => dataset === 'non_epr_fields.test_metrics_2'
       );
+      // These are tests for source_mode set in the package. Couldn't find any integration tests around the policy part?
+      // Is this all covered by unit tests? I'm especially curious how an upgrade around this feature is tested?
+      // And how do you test it manually with uploading a package when you run from source -> run your own registry?
```

Comment:
> These are tests for source_mode set in the package. Couldn't find any integration tests around the policy part?

I don't think we have integration tests for the toggling functionality/API interactions here, so those would probably be good to add. 

We do have unit tests around the policy generation in 
- https://github.com/elastic/kibana/blob/86527753a2db307138ea69e0ba04551717a91844/x-pack/plugins/fleet/common/services/package_to_package_policy.test.ts
- https://github.com/elastic/kibana/blob/main/x-pack/plugins/fleet/common/services/simplified_package_policy_helper.test.ts

There's UI driven tests around the toggling that assert on whether we're generating the expected API payloads, defaulting to the expected values based on the existing data streams, etc here as well: https://github.com/elastic/kibana/blob/86527753a2db307138ea69e0ba04551717a91844/x-pack/plugins/fleet/public/applications/fleet/sections/agent_policy/create_package_policy_page/components/steps/components/experimental_datastream_settings.test.tsx

> And how do you test it manually with uploading a package when you run from source -> run your own registry?

We create a new version of package that includes time series features, copy that new version over to a data directory, then point the local registry at that data directory. Then, the test package version is available as normal in Kibana, Fleet API, etc for testing upgrades. We don't use the upload package feature in this case. The registry service takes a config that includes locations of package manifests on disk, so we can't point it at a few directories that it'll use to source packages.